ccl/changefeedccl: TestTableNameCollision failed

ccl/changefeedccl.TestTableNameCollision [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/18912936?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/18912936?buildTab=artifacts#/) on release-23.2 @ [333a1ca8e2803b38d71278a4d9f37144ebf29db9](https://github.com/cockroachdb/cockroach/commits/333a1ca8e2803b38d71278a4d9f37144ebf29db9):

Fatal error:

```
panic: use of Span after Finish. Span: changefntr. Finish previously called at: goroutine 2021533 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x5e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).finishInternal(0xc004abfc00)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:251 +0x67
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).Finish(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:242
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).InternalClose(0xc010713800)
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:918 +0xb5
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeFrontier).close(0xc010713800)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:1442 +0x45
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*changeFrontier).ConsumerClosed(0xc00f134e00?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_processors.go:1901 +0x13
github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run({0x7db5af8, 0xc006dac000}, {0x7dd7240, 0xc010713800}, {0x7d7efe0?, 0xc00f134e00?})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/base.go:204 +0xa4
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc010713800, {0x7db5ac0?, 0xc01f31f7c0?}, {0x7d7efe0?, 0xc00f134e00})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:721 +0x6d
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc00bdcf900, {0x7db5ac0?, 0xc01f31f7c0}, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:574 +0x222
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc0164f9400, {0x7db5ac0, 0xc01f31f720}, 0xc011d2b440, 0x0, 0xc007804b80, 0xc00f134e00, 0xc010713000, 0xc0126a9f18)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:914 +0xb9e
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.startDistChangefeed.func1({0x7db5ac0, 0xc01f31f5e0})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist.go:316 +0x408
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GoAndWait.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:163 +0x22
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 2021022
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x96


```
Stack: 

```
goroutine 2021345 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xbe9a660?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:177 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc004abfc00)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:437 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:445
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x7db5ac0, 0xc00b83d360})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:87 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x7db5ac0, 0xc00b83d360}, 0x2, 0x3, 0x0, 0x3?, {0x66e597a, 0x21}, {0xc00a9e9ca0, 0x1, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:99 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:34
github.com/cockroachdb/cockroach/pkg/util/log.Errorf(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:654
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*batchingSink).runBatchingWorker(0xc01608e0e0, {0x7db5ac0?, 0xc00b83d360})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/batching_sink.go:434 +0x5cf
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeBatchingSink.func1({0x7db5ac0?, 0xc00b83d360?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/batching_sink.go:544 +0x2a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeBatchingSink.Group.GoCtx.func2()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:163 +0x22
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 2021533
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x96
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTableNameCollision
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestTableNameCollision1206430527
    test_log_scope.go:76: use -show-logs to present logs inline
=== RUN   TestTableNameCollision/kafka
    helpers_test.go:873: making server as system tenant
    helpers_test.go:948: making kafka feed factory
    testfeed_test.go:300: creating external connection
    testfeed_test.go:303: ran create external connection
    testfeed_test.go:300: creating external connection
    testfeed_test.go:303: ran create external connection
    testfeed_test.go:300: creating external connection
    testfeed_test.go:303: ran create external connection
```
</p>
</details>

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTableNameCollision.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-48074

--- a/pkg/ccl/changefeedccl/batching_sink.go
+++ b/pkg/ccl/changefeedccl/batching_sink.go
@@ -17,10 +17,8 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/intsets"
-	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
-	"github.com/cockroachdb/errors"
 )
 
 // SinkClient is an interface to an external sink, where messages are written
@@ -429,11 +427,12 @@ func (s *batchingSink) runBatchingWorker(ctx context.Context) {
 	for {
 		select {
 		case req := <-s.eventCh:
-			if err := s.pacer.Pace(ctx); err != nil {
-				if !errors.Is(err, context.Canceled) && pacerLogEvery.ShouldLog() {
-					log.Errorf(ctx, "automatic sink batcher pacing: %v", err)
-				}
-			}
+			// Swallow pacer error -- it happens only if context is canceled,
+			// and that's handled below.
+			// TODO(yevgeniy): rework this function: this function should simply
+			// return an error, and not rely on "handleError".
+			// It's hard to reason about this functions correctness otherwise.
+			_ = s.pacer.Pace(ctx)
 
 			switch r := req.(type) {
 			case *rowEvent:

--- a/pkg/ccl/changefeedccl/event_processing.go
+++ b/pkg/ccl/changefeedccl/event_processing.go
@@ -10,7 +10,6 @@ import (
 	"hash"
 	"hash/crc32"
 	"runtime"
-	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval"
 	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdcevent"
@@ -33,10 +32,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 )
-
-// pacerLogEvery is used for logging errors instead of returning terminal
-// errors when pacer.Pace returns an error.
-var pacerLogEvery log.EveryN = log.Every(100 * time.Millisecond)
 
 // eventContext holds metadata pertaining to event.
 type eventContext struct {
@@ -325,9 +320,7 @@ func (c *kvEventToRowConsumer) ConsumeEvent(ctx context.Context, ev kvevent.Even
 	// unavailable. If there is unused CPU time left from the last call to
 	// Pace, then use that time instead of blocking.
 	if err := c.pacer.Pace(ctx); err != nil {
-		if pacerLogEvery.ShouldLog() {
-			log.Errorf(ctx, "automatic pacing: %v", err)
-		}
+		return err
 	}
 
 	schemaTimestamp := ev.KV().Value.Timestamp

